content/tutorials/08-grafana.md

```diff
@@ -0,0 +1,223 @@
+
+You should see something similar to the following:
+
+::: div image-xl
+<svg title='' src='throughput.png'/>
+:::
+
+4. Select **Apply** to save the panel.
+
+### Time series graph of HTTP throughput per service
+
+1. Create a new panel following the directions in the previous section.
+
+2. Copy & paste this query into the Query Editor.
+
+```python
+# Import Pixie's module for querying data.
+import px
+
+# Load data from Pixie's `http_events` table into a Dataframe.
+df = px.DataFrame(table='http_events', start_time=__time_from)
+
+# Add K8s metadata context.
+df.service = df.ctx['service']
+df.namespace = df.ctx['namespace']
+
+# Bin the 'time_' column using the interval provided by Grafana.
+df.timestamp = px.bin(df.time_, __interval)
+
+# Group data by unique pairings of 'timestamp' and 'service'
+# and count the total number of requests per unique pairing.
+per_ns_df = df.groupby(['timestamp', 'service']).agg(
+        throughput_total=('latency', px.count)
+    )
+
+# Calculate throughput by dividing # of requests by the time interval.
+per_ns_df.request_throughput = per_ns_df.throughput_total / __interval
+per_ns_df.request_throughput = per_ns_df.request_throughput * 1e9
+
+# Rename 'timestamp' column to 'time_'. The Grafana plugin expects a 'time_'
+# column to display data in a Graph or Time series.
+per_ns_df.time_ = per_ns_df.timestamp
+
+# Output select columns of the DataFrame.
+px.display(per_ns_df['time_', 'service', 'request_throughput'])
+```
+
+This PxL query outputs a table of timeseries data for HTTP request throughput per service
+
+3. On the Panel tab, under the **Visualization** drop-down menu, select **Time series**.
+
+4. Add units to the graph on the **Field** tab, under the **Standard options** drop-down menu, under the **Unit** menu, select **requests/sec**.
+
+You should see something similar to the following:
+
+::: div image-xl
+<svg title='' src='service-throughput.png'/>
+:::
+
+5. Select **Apply** to save the panel.
+
+### Bar chart of HTTP errors per service
+
+The following PxL query outputs a table of HTTP error count per service.
+
+1. Create a new panel following the directions in the previous section.
+
+2. Copy & paste this query into the Query Editor.
+
+```python
+import px
+
+# Import HTTP events table.
+df = px.DataFrame(table='http_events', start_time='-5m')
+
+# Add columns for service, namespace info.
+df.namespace = df.ctx['namespace']
+df.service = df.ctx['service']
+
+# Filter out requests that don't have a service defined.
+df = df[df.service != '']
+
+# Filter out requests from the Pixie (pl) namespace.
+df = df[df.namespace != 'pl']
+
+# Add column for HTTP response status errors.
+df.error = df.resp_status >= 400
+
+# Group HTTP events by service, counting errors and total HTTP events.
+df = df.groupby(['service']).agg(
+    error_count=('error', px.sum),
+    total_requests=('resp_status', px.count)
+)
+
+# Output the DataFrame.
+px.display(df)
```

Comment:
Did you want to include the multiple display calls/multiple tables can be returned or is that extraneous information?